lib/pbench/server/cache_manager.py

```diff
@@ -226,6 +255,76 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
         return cls(destination, controller)
 
+    def cache_map(self, tar_path: Path):
+        """Builds Hierarchy structure of a Directory in a Dictionary
+        Format.
+
+        Args:
+            dir_path: root of a directory
+        """
+        cmap = {tar_path.name: {"details": FileInfo(tar_path)}}
+        dir_queue = deque([(tar_path, cmap)])
+        while dir_queue:
+            tar_path, parent_map = dir_queue.popleft()
+            tar_n = tar_path.name
+            curr = parent_map[tar_n]
+
+            dir_list = list(tar_path.glob("*"))
+            if dir_list:
+                curr["children"] = {}
+                curr = curr["children"]
+
+            for l_path in dir_list:
+                tar_info = FileInfo(l_path)
+                curr[l_path.name] = {"details": tar_info}
+                if l_path.is_dir():
+                    dir_queue.append((l_path, curr))
+        self.cachemap = cmap
+
+    @staticmethod
+    def traverse_cmap(path: Path, cachemap: dict):
+        name = path.name
+        files_list = path.parts
+        c_map = cachemap
+
+        for file_l in files_list:
+            if file_l not in ("/", "//"):
+                c_map = c_map[file_l]
+                if c_map["details"].type == "directory":
```

Comment:
Similarly here, if we find that `/var` is not a `"directory"`, `c_map` will "stick" there and that's what we'll return.

(NOTE, an Enum for node "types" would be much less error prone than using strings; even if we're thinking about capturing our JSON outside of Python, for example in Redis, where we can't use non-string values, using an Enum `DIRECTORY.name` or even a string constant would reduce the risk of errors.)